### ISSUE TITLE
Too many requests error

### ISSUE THREAD

**Issue by chartersl** [microsoft/pxt-microbit#3250]
Hi 

We're a school with close to 400 students attempting microbit projects. 

We have embedded code modules into a webpage so they can access lessons from home or school.
When a class attempts to access the page at school, the modules quickly stop loading for students due to 429 "Too Many Requests" errors.

It appears this happened in the past for people per issue #706.
https://github.com/Microsoft/pxt-microbit/issues/706

We also have been experiencing the "Network Request Failed!" error on iPads when attempting to install the Robit extension. The first few students in a class can install it and then the rest receive the error.

I'm assume both issues stem from the fact that as we are a school, all the students exit the firewall appearing from a small range of IP addresses and your DDoS protection kicks in with the volume of requests received from them. 

Are you able to ease the restriction on the number of requests your servers receive from a single IP or could we provide you with our outbound IP address range and you can whitelist them?

Kind regards,
Luke

**Comment by abchatra** [comment]
@chartersl  thanks for the report. Can you share the embedded code modules?

**Comment by chartersl** [comment]
Here you go:

https://makecode.microbit.org/#pub:_hwJ0XF3agA3F
https://makecode.microbit.org/#pub:_HsJ9JdhwYhhR
https://makecode.microbit.org/#pub:_U81Ffa7iMi9f
https://makecode.microbit.org/#pub:_aAEPjiDj8Fk8
https://makecode.microbit.org/#pub:_HWqDRmM73TVP
https://makecode.microbit.org/#pub:_boWE4caF9CWr
https://makecode.microbit.org/#pub:_J3vTW4Toq2Kg
https://makecode.microbit.org/#pub:_UYLEuJAiR8a5
https://makecode.microbit.org/#pub:_bJDWwACrKP2E
https://makecode.microbit.org/#pub:_1v3AAY0zedAb
https://makecode.microbit.org/#pub:_Ds4HTbii2a5C
https://makecode.microbit.org/#pub:_TmteCEH79Uu7
https://makecode.microbit.org/#pub:_Yk2JDpJzHFxe

**Comment by abchatra** [comment]
@chartersl I looked at the snippets, its likely due to pxt-robit extension. As they are used in many of the embedded code modules as well.

We have just released and update to microbit on June 12th. We addressed many of the network throttling issues, although I am unsure if we have addressed this particular issue.

Is it possible to see if the latest update fixed these?  I will continue investigation on our side as well.
